### ISSUE TITLE
Silent failures when BIGTABLE_EMULATOR_HOST is set

### ISSUE THREAD

**Issue by vitaminmoo** [hashicorp/terraform-provider-google#9173]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

v0.15.3

### Affected Resource(s)

* google_bigtable_*

### Terraform Configuration Files

```tf
provider "google" {
 project     = "XXX"
}
resource "google_bigtable_instance" "bigtable1" {
  name = "bigtable1"
  cluster {
    cluster_id   = "bigtable1-cluster"
    zone         = "us-central1-f"
    num_nodes    = 1
    storage_type = "SSD"
  }
  lifecycle {
    prevent_destroy = true
    ignore_changes = [
      cluster[0].num_nodes
    ]
  }
}
```

### Debug Output

https://gist.github.com/vitaminmoo/e2055a8bce6bdc5a1515c99fb71ff3be

Note that this output is from `terraform import`, which is where I found the issue. Based on how the bigtable client works, it very likely will break other bigtable resource actions too. Additionally, the issue will only happen if the bigtable emulator is running locally.

The line making my specific error invisible is https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_bigtable_instance.go#L215

`err` in this case is `rpc error: code = Unimplemented desc = unimplemented feature`

### Panic Output

No panic

### Expected Behavior

The bigtable instance should have been imported, or a warning should have been printed about how we're not connecting to real bigtable.

### Actual Behavior

Success is reported, but the bigtable instance does not show up in the state file

### Steps to Reproduce

1. Configure bigtable emulator locally including setting $BIGTABLE_EMULATOR_HOST
1. `terraform import` a bigtable that exists in a real project
1. Observe output/results

### Important Factoids

N/A

### References

https://github.com/GoogleCloudPlatform/magic-modules/pull/2927

**Comment by rileykarson** [comment]
Can you link documentation on running the emulate so I can attempt to repro this and see why import is failing? Terraform makes it difficult to display error messages to users, so I may need to resort to logging the error in debug logs otherwise which won't help much.

**Comment by vitaminmoo** [comment]
Here you go: https://cloud.google.com/bigtable/docs/emulator

**Comment by vitaminmoo** [comment]
Note that the bigtable emulator does not support the InstanceInfo call, so even if one was purposefully pointing at the emulator, the import would not work (some other operations likely would not as well).

In my case, we use the emulator for other testing, so the env variable was accidentally causing imports to fail.

I definitely consider the env var being set for terraform to be a configuration error on my end, it just took a rather inordinate amount of time to figure out what the issue was, due to the silent failure and masking of the error returned at https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_bigtable_instance.go#L215

**Comment by kevinsi4508** [comment]
@telpirion FYI.

@rileykarson , I thought import should fail if the resource is not found (ID is cleared).

According this log `2021-05-18T10:16:22.995-0700 [INFO]  provider.terraform-provider-google_v3.67.0_x5: 2021/05/18 10:16:22 [WARN] Removing bigtable1 because it's gone: timestamp=2021-05-18T10:16:22.995-0700`, the resource was cleared/removed.

I believe this might require Provider code first, not Bigtable specific logic.

**Comment by rileykarson** [comment]
I think this is happening for the same reason as you've been investigating already @kevinsi4508! We were setting the `id` to `""` too often, and that caused a silent failure here.

**Comment by kevinsi4508** [comment]
Right, we shouldn't set `id` to `""` too often, but shouldn't import fails if the `id` is set to `""` in this case? It almost sounds like I want to import X, but we couldn't do it; instead we clear the whole thing. This means the import failed, isn't it?

**Comment by rileykarson** [comment]
Yeah, it should fail in cases where it's `""` too. I'm not sure why it wouldn't.

**Comment by kevinsi4508** [comment]
The import result might be depending on the status returned, which is no error: "nil".

Maybe there should be a check before we declare import is a success?

1. Import
2. Check id is not empty or check no drift right after import?

**Comment by rileykarson** [comment]
Import's handled through the provider framework based on a combination of our import func (https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/resource_importer.go) and Read func, we don't have much opportunity to fine-tune it. See https://learn.hashicorp.com/tutorials/terraform/provider-import?in=terraform/providers for an example.

Something's going wrong _somewhere_ such that we're getting a silent failure, I'm just not sure where offhand.

**Comment by melinath** [comment]
b/259277173

**Comment by CodingAnarchy** [comment]
I just ran into this issue earlier today, though instead of a silent failure it now fails very loudly that the RPC is unimplemented. It does seem like the TF provider should ignore this environment variable while it is running - can it not be unset inside the TF process itself?
